database/src/main/java/database/Main.java

```diff
@@ -26,11 +28,12 @@ public static void main(String[] args) {
                             }))));
             config.registerPlugin(new SwaggerPlugin());
 
-            final var databaseService = new SqlDatabaseService(settings);
             final var dbServiceRestController = new DatabaseServiceRestController(databaseService, entities);
             dbServiceRestController.configureRouter(config);
         }).start(port);
 
+        app.events(event -> event.serverStopping(databaseService::shutdown));
+
         var databaseControllerExceptionHandler = new DatabaseControllerExceptionHandler();
         databaseControllerExceptionHandler.register(app);
```

Comment:
Please add shutdown signal after this string to properly finish Javalin application and trigger `event.serverStopping(databaseService::shutdown))` method. You can use the following code:
```
Runtime.getRuntime().addShutdownHook(new Thread(() -> {
    System.out.println("Shutdown hook triggered. Stopping Javalin...");
        app.stop();
    }));
```